src/MediaWiki.ts

```diff
@@ -11,26 +11,126 @@ import basicURLDirector from './util/builders/url/basic.director.js'
 import BaseURLDirector from './util/builders/url/base.director.js'
 import ApiURLDirector from './util/builders/url/api.director.js'
 
+interface MWCapabilities {
+  apiAvailable: boolean
+  veApiAvailable: boolean
+  coordinatesAvailable: boolean
+  desktopRestApiAvailable: boolean
+}
 class MediaWiki {
   public metaData: MWMetaData
   public readonly baseUrl: URL
-  public readonly modulePath: string
-  public readonly webUrl: URL
-  public readonly apiUrl: URL
-  public readonly veApiUrl: URL
-  public readonly restApiUrl: URL
-  public readonly mobileRestApiUrl: URL
-  public readonly desktopRestApiUrl: URL
+  public readonly modulePathConfig
   public readonly getCategories: boolean
   public readonly namespaces: MWNamespaces = {}
   public readonly namespacesToMirror: string[] = []
 
   private readonly wikiPath: string
+  private readonly restApiPath: string
   private readonly username: string
   private readonly password: string
   private readonly apiPath: string
   private readonly domain: string
   private apiUrlDirector: ApiURLDirector
+  private baseUrlDirector: BaseURLDirector
+
+  private _veapiUrl: URL
+  private _restApiUrl: URL
+  private _apiUrl: URL
+  private _modulePath: string
+  private _webUrl: URL
+  private _desktopRestApiUrl: URL
+  // TODO: Mobile builder was removed since there is no /mobile-sections endpoint
+
+  // Set default MW capabilities
+  private readonly mwCapabilities: MWCapabilities
+
+  /**
+   * veApiUrl based on top of 'new ApiURLDirecto'
+   */
+  public get veapiUrl(): URL {
+    if (!this._veapiUrl) {
+      // TODO: This depend on baseUrlDirector.buildURL(this.apiPath) and looks like a weak solution
+      this._veapiUrl = this.apiUrlDirector.buildVisualEditorURL()
+    }
+    return this._veapiUrl
```

Comment:
Not easy, considering that you have so many different styles and developers behind the codebase... but think about:
* `Url` vs `URL`
* `VE` vs `VisualEditor`.

Reply:
I'd like that we have a dedicated ticket about naming conventions.

Reply:
For VE, the answer is in the validation of https://github.com/openzim/mwoffliner/wiki/API-end%E2%80%90points. For `URL`, I clearly prefer to keep full uppercase for constants but up to you.

Reply:
I renamed `ve` to `visualEditor` but regarding `Url` I'd rather keep it as is because there are about ~580 entries across mwoffliner that use the same capitalization.